numpy/core/src/umath/loops.c.src

```diff
@@ -1683,7 +1684,7 @@ pairwise_sum_@TYPE@(char *a, npy_uintp n, npy_intp stride)
 
         for (i = 8; i < n - (n % 8); i += 8) {
             /* small blocksizes seems to mess with hardware prefetch */
-            NPY_PREFETCH(a + (i + 512 / sizeof(@dtype@)) * stride, 0, 3);
+            NPY_PREFETCH(a + (i + 512 / (int)sizeof(@dtype@)) * stride, 0, 3);
```

Comment:
Shouldn't this cast be `intp`?

Reply:
Just needs to be a signed type big enough to hold `sizeof(@dtype@)`, which is certainly true here. Mostly, I wanted to keep the line short ;)

Reply:
Note that the literal `512` is also of type `int`.

Reply:
It compiles as an immediate as expected. For instance, putting it in a function and using double as the type
```
movl    $64, %eax
```
Where `eax` is the return register.

Reply:
My point is that we risk overflow if we ever allow dtypes larger than an int. 512 would undergo integer promotion to size_t, so it's type is not relevant.

Reply:
The loops here are generated for floating/complex types, so not a problem.

Reply:
The point is that the result needs to be signed as it is added to `i` and multiplied by `stride`, which is `npy_intp`. It may be that `stride` is always positive, but I haven't convinced myself that that is the case.

Reply:
I'm advicating the cast becoming `(npy_intp)`. I don't see a float type with a `sizeof` too large for an `int` in my lifetime, so I suppose what you have is fine ;)

Reply:
I suppose you could argue on the principle of least surprise.